packages/manager/modules/backup-agent/src/data/hooks/agents/putAgent.ts

```diff
@@ -0,0 +1,40 @@
+import { UseMutationOptions, useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+
+import {
+  EditConfigurationBackupAgentsProps,
+  editConfigurationBackupAgents,
+} from '@/data/api/agents/agents.requests';
+
+import { GET_VSPC_TENANTS_QUERY_KEY } from '../tenants/useVspcTenants';
+
+type UseEditConfigurationVSPCTenantAgentParams = Partial<
+  UseMutationOptions<
+    ApiResponse<string>,
+    ApiError,
+    Omit<EditConfigurationBackupAgentsProps, 'vspcTenantId' | 'backupAgentId'>
+  >
+>;
+
+export const useEditConfigurationVSPCTenantAgent = ({
+  vspcTenantId,
+  backupAgentId,
+  ...options
+}: UseEditConfigurationVSPCTenantAgentParams &
+  Pick<EditConfigurationBackupAgentsProps, 'vspcTenantId' | 'backupAgentId'>) => {
```

Comment:
```suggestion
type UseEditConfigurationVSPCTenantAgentParams = Pick<EditConfigurationBackupAgentsProps, 'vspcTenantId' | 'backupAgentId'>
& Partial<
  UseMutationOptions<
    ApiResponse<string>,
    ApiError,
    Omit<EditConfigurationBackupAgentsProps, 'vspcTenantId' | 'backupAgentId'>
  >
> ;

export const useEditConfigurationVSPCTenantAgent = ({
  vspcTenantId,
  backupAgentId,
  ...options
}: UseEditConfigurationVSPCTenantAgentParams ) => {
```
you can probably manage your hook params types in the same declaration, it'll be easier to read.